src/formatDate.js

```diff
@@ -48,9 +48,61 @@
  *
  * @returns {string}
  */
+const CENTURY_21 = '20';
+const CENTURY_20 = '19';
+const MONTH_ACRONYM = 'MM';
+const DAY_ACRONYM = 'DD';
+const SHORT_YEAR_ACRONYM = 'YY';
+const LONG_YEAR_ACRONYM = 'YYYY';
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const expectedDate = [];
+  let month, day, shortYear, longYear;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case MONTH_ACRONYM:
+        month = dateArr[i];
+        break;
+      case DAY_ACRONYM:
+        day = dateArr[i];
+        break;
+      case SHORT_YEAR_ACRONYM:
+        shortYear = dateArr[i];
+        break;
+      case LONG_YEAR_ACRONYM:
+        longYear = dateArr[i];
+        break;
+    }
+  }
+
+  if (!shortYear) {
+    shortYear = longYear.slice(2, 4);
+  }
```

Comment:
No need to change - but an even better idea will be combining conditions with switch block
```suggestion
  for (let i = 0; i < fromFormat.length; i++) {
    switch (fromFormat[i]) {
      case MONTH_ACRONYM:
        month = dateArr[i];
        break;
      case DAY_ACRONYM:
        day = dateArr[i];
        break;
      case SHORT_YEAR_ACRONYM:
        shortYear = dateArr[i];
        break;
      case LONG_YEAR_ACRONYM:
        longYear = dateArr[I];
        shortYear = longYear.slice(2, 4);
        break;
    }
  }

```
And same for the next condition with long year